helpers/foundation-deployer/stages/apply.go

```diff
@@ -0,0 +1,284 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package stages
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/mitchellh/go-testing-interface"
+
+	"github.com/terraform-google-modules/terraform-example-foundation/helpers/foundation-deployer/gcp"
+	"github.com/terraform-google-modules/terraform-example-foundation/helpers/foundation-deployer/msg"
+	"github.com/terraform-google-modules/terraform-example-foundation/helpers/foundation-deployer/steps"
+	"github.com/terraform-google-modules/terraform-example-foundation/helpers/foundation-deployer/utils"
+)
+
+
+func DeployBootstrapStage(t testing.TB, s steps.Steps, tfvars GlobalTFVars, c CommonConf) error {
+	repo := "gcp-bootstrap"
+	step := "0-bootstrap"
+
+	bootstrapTfvars := BootstrapTfvars{
+		OrgID:              tfvars.OrgID,
+		DefaultRegion:      tfvars.DefaultRegion,
+		BillingAccount:     tfvars.BillingAccount,
+		GroupOrgAdmins:     tfvars.GroupOrgAdmins,
+		GroupBillingAdmins: tfvars.GroupBillingAdmins,
+		ParentFolder:       tfvars.ParentFolder,
+		ProjectPrefix:      tfvars.ProjectPrefix,
+		FolderPrefix:       tfvars.FolderPrefix,
+		BucketForceDestroy: tfvars.BucketForceDestroy,
+	}
+
+	err := utils.WriteTfvars(filepath.Join(c.FoundationPath, step, "terraform.tfvars"), bootstrapTfvars)
+	if err != nil {
+		return err
+	}
+
+	terraformDir := filepath.Join(c.FoundationPath, step)
+	options := &terraform.Options{
+		TerraformDir: terraformDir,
+		Logger:       c.Logger,
+		NoColor:      true,
+	}
+	// terraform deploy
```

Comment:
nit:  I suspect we could move this to a func since the init-plan-apply is going to be the same for most stages?

Reply:
local init-plan-apply will be also used for the local deploy of the shared environments in the network and projects stages.
nove to a function `applyLocal`